vit-servicing-station-tests/src/tests/rest/voteplan_id.rs

```diff
@@ -0,0 +1,49 @@
+use crate::common::data::valid_fund_snapshot;
+use crate::common::startup::{db::DbBuilder, server::ServerBootstrapper};
+use assert_fs::TempDir;
+use vit_servicing_station_lib::v0::endpoints::proposals::ProposalVoteplanIdAndIndexes;
+
+#[test]
+pub fn get_proposals_by_voteplan_id_and_index() {
+    let temp_dir = TempDir::new().unwrap().into_persistent();
+
+    let snapshot = valid_fund_snapshot();
+
+    let db_path = DbBuilder::new()
+        .with_snapshot(&snapshot)
+        .build(&temp_dir)
+        .unwrap();
+
+    let server = ServerBootstrapper::new()
+        .with_db_path(db_path.to_str().unwrap())
+        .start(&temp_dir)
+        .unwrap();
+
+    let rest_client = server.rest_client_with_token(&snapshot.token_hash());
+    let expected_proposals = rest_client.proposals().unwrap();
+
+    let vote_plan_id: String = snapshot.voteplans()[0].chain_voteplan_id.clone();
+    let indexes: Vec<i64> = expected_proposals
+        .iter()
+        .filter(|x| x.chain_voteplan_id == vote_plan_id)
+        .map(|p| p.chain_proposal_index)
+        .collect();
+
+    let actual_proposals = rest_client
+        .proposals_by_voteplan_id_and_index(vec![ProposalVoteplanIdAndIndexes {
+            vote_plan_id,
+            indexes: indexes.clone(),
+        }])
+        .unwrap();
+
+    assert!(actual_proposals.len() > 0);
+
+    for proposal in actual_proposals {
+        let expected_proposal = expected_proposals
+            .iter()
+            .find(|x| proposal.proposal.internal_id == x.internal_id)
+            .unwrap();
+
+        assert_eq!(&proposal.proposal, expected_proposal);
```

Comment:
```suggestion
    expected_proposals.sort_by(|p| p.proposal_id);
    actual_proposals.sort_by(|p| p.proposal_id);
    assert_eq!(actual_proposals, expected_proposals);
```

Reply:
I think it won't work because actual_proposals and expected_proposals have different types

Reply:
I guess his suggestion is to avoid the quadratic part. In that case you could do an extra loop on `expected_proposals.iter().zip(actual_proposals.iter())` and call `assert_eq` for each one, and it would be the same.

Or `expected_proposals` could be a `HashMap`.

I wonder if it really matters though.

Reply:
ok, right. implemented :)

Reply:
It was not just for inefficiency, the current method does not check that every expected proposal is in `actual_proposals`